### PR TITLE
feat(agents): add security-reviewer subagent with improved tests

### PR DESCRIPTION
**Pull request by jpoley**
## Summary
- Add security-reviewer subagent definition for security validation workflows
- Comprehensive test suite with defensive coding patterns
- High-quality code with type hints, docstrings, and proper error handling

## Key Improvements
- **Type Safety**: Full type hints on all functions and methods
- **Documentation**: Detailed docstrings with Args, Returns, Raises sections
- **Error Handling**: UTF-8 encoding on all file operations, proper exception handling
- **Code Quality**: Helper functions to reduce duplication (read_file_safely, parse_frontmatter, get_tools_list)
- **Better Testing**: Meaningful assertion messages that explain failures
- **Defensive Coding**: Input validation and edge case handling

## Test Coverage
- 33 security-reviewer agent tests (all passing)
- Frontmatter validation (YAML structure, required fields)
- Tool configuration (read-only tools, no Write/Edit)
- Content validation (OWASP, CWE, SLSA, security best practices)
- File integrity (agent and template must match)

## Test plan
- [x] Agent definition file present and valid
- [x] All 33 agent tests pass
- [x] Full test suite passes (1758 tests)
- [x] Lint passes (ruff check --fix)
- [x] Format passes (ruff format)
- [x] DCO sign-off included

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: templates/agents/security-reviewer.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+from pydantic import BaseModel, Field, validator
+
+class UserInput(BaseModel):
+    email: str = Field(..., pattern=r"^[\w\.-]+@[\w\.-]+\.\w+$")
+    age: int = Field(..., ge=0, le=150)
+
+    @validator('email')
+    def email_domain_allowed(cls, v):
+        allowed = ['company.com', 'partner.com']
+        domain = v.split('@')[1]
```
[the remark this answers is not in the document]
Comment (jpoley) [reply templates/agents/security-reviewer.md]: fix me

[the remark this answers is not in the document]
Comment (jpoley) [reply templates/agents/security-reviewer.md]: fix me

File: .claude/agents/security-reviewer.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
+---
+name: security-reviewer
+description: Use this agent for security tasks including vulnerability assessment, code security review, threat modeling, SLSA compliance, and security testing. Examples: <example>Context: User needs a security review. user: "Review the authentication module for security issues" assistant: "I'll use the security-reviewer agent to perform a thorough security review of the authentication module." <commentary>Security reviews should use the security-reviewer agent for specialized expertise.</commentary></example> <example>Context: User wants to check for vulnerabilities. user: "Are there any SQL injection vulnerabilities in our API?" assistant: "Let me use the security-reviewer agent to analyze the API for SQL injection vulnerabilities." <commentary>Vulnerability assessment requires security-reviewer expertise.</commentary></example>
+tools: Read, Glob, Grep, Bash
```
[the remark this answers is not in the document]
Comment (jpoley) [reply .claude/agents/security-reviewer.md]: fix me

**Comment by jpoley** [comment]
fixes needed
